enterprise/reporting/setup.go

```diff
@@ -12,43 +12,63 @@ import (
 )
 
 type Factory struct {
-	EnterpriseToken   string
-	Log               logger.Logger
-	once              sync.Once
-	reportingInstance types.ReportingI
+	EnterpriseToken     string
+	Log                 logger.Logger
+	once                sync.Once
+	reportingInstance   types.ReportingI
+	edReportingInstance types.ReportingI
 }
 
 // Setup initializes Suppress User feature
-func (m *Factory) Setup(backendConfig backendconfig.BackendConfig) types.ReportingI {
+func (m *Factory) Setup(backendConfig backendconfig.BackendConfig) types.ReportingInstances {
 	if m.Log == nil {
 		m.Log = logger.NewLogger().Child("enterprise").Child("reporting")
 	}
 	m.once.Do(func() {
 		reportingEnabled := config.GetBool("Reporting.enabled", types.DefaultReportingEnabled)
 		if !reportingEnabled {
 			m.reportingInstance = &NOOP{}
+			m.edReportingInstance = &NOOP{}
 			return
 		}
 
 		if m.EnterpriseToken == "" {
 			m.reportingInstance = &NOOP{}
+			m.edReportingInstance = &NOOP{}
 			return
 		}
 
 		h := NewFromEnvConfig(m.Log)
+		ed := NewEdReporterFromEnvConfig()
 		rruntime.Go(func() {
 			h.setup(backendConfig)
 		})
+		rruntime.Go(func() {
+			ed.setup(backendConfig)
+		})
 		m.reportingInstance = h
+		m.edReportingInstance = ed
 	})
-
-	return m.reportingInstance
+	return types.ReportingInstances{
+		ReportingInstance:   m.reportingInstance,
+		EdReportingInstance: m.edReportingInstance,
+	}
 }
 
-func (m *Factory) GetReportingInstance() types.ReportingI {
-	if m.reportingInstance == nil {
-		panic(fmt.Errorf("reporting instance not initialised. You should call Setup before GetReportingInstance"))
+func (m *Factory) GetReportingInstance(names ...string) types.ReportingI {
```

Comment:
since it is internal it is better to change signature of GetReportingInstance to GetReportingInstance(name string) as current signature is hacky

Reply:
It is not internal method.
GetReportingInstance is being used by embeddedAppHandler & processAppHandler.

Reply:
Internal I mean, we can control, don't need to worry about backward compatibility

Reply:
Ok. Will make relevant changes